skyportal/models.py

```diff
@@ -3129,6 +3130,25 @@ class Thumbnail(Base):
         nullable=False,
         doc="ID of the thumbnail's obj.",
     )
+    is_grayscale = sa.Column(
+        sa.Boolean(),
+        nullable=False,
+        default=False,
+        doc="Boolean indicating whether the thumbnail is (mostly) grayscale or not.",
+    )
+
+
+@event.listens_for(Thumbnail, 'before_insert')
+def classify_thumbnail_grayscale(mapper, connection, target):
+    if target.file_uri is not None:
+        target.is_grayscale = image_is_grayscale(target.file_uri)
+    else:
+        try:
+            target.is_grayscale = image_is_grayscale(
+                requests.get(target.public_url, stream=True).raw
+            )
+        except requests.exceptions.RequestException:
+            pass
```

Comment:
Have you tested that the front-end doesn't error when `is_grayscale` is null?

Reply:
Ah no, not specifically but I don't think that should ever be the case because I've set the data model column to have `nullable=False` with `default=False`. Basically, all thumbnails will be assumed to be colored by default to start out until the event listener to classify it runs. The `try...except...` here will catch errors if external URL thumbnails (like PS1) lead to bad requests for whatever reason and those thumbnails will just keep the default `is_grayscale  = False`.

Reply:
Sounds good